app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+
+    if (this.isNewPage) {
+      this.module.addEventListener('click', this.handleClick.bind(this))
+    } else {
+      if (window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker) {
+        var pageViewTracker = window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker
+        window.dataLayer.push({
+          event: 'page_view',
+          page_view: {
+            location: pageViewTracker.getLocation(),
+            referrer: pageViewTracker.getReferrer()
+          }
+        })
+      }
+    }
+    
+    this.trackSearchResults()
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.trackSearchResults = function () {
+    var schema = this.populateEcommerceSchema(this.module)
+    this.clearPreviousEcommerceObject()
+    window.dataLayer.push(schema)
+  }
+
+  // the handleClick function will listen for clicks on search results and push to the dataLayer every time a search result is clicked
+  Ga4EnhancedEcommerceTracker.prototype.handleClick = function (event) {
+    var isSearchResult = event.target.getAttribute('data-ecommerce-path')
+
+    if (isSearchResult) {
+      var searchResult = event.target
+      var schema = this.populateEcommerceSchema(true, searchResult)
+
+      this.clearPreviousEcommerceObject()
+      window.dataLayer.push(schema)
+    }
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.populateEcommerceSchema = function (searchResultClicked = false, searchResult = null) {
+    // Limiting to 100 characters to avoid noise from extra longs search queries
+    // and to stop the size of the payload going over 8k limit.
+    var searchQuery = this.PIIRemover.stripPII(this.module.getAttribute('data-search-query')).substring(0, 100).toLowerCase()
+    var variant = this.module.getAttribute('data-ecommerce-variant') || undefined
+    var ecommerceRows = this.module.querySelectorAll('[data-ecommerce-row]')
+    var listTitle = this.module.getAttribute('data-list-title') || this.DEFAULT_LIST_TITLE
+    var startPosition = parseInt(this.module.getAttribute('data-ecommerce-start-index'), 10)
+    var ecommerceObject = {}
+
+    ecommerceObject.event = 'search_results'
+    ecommerceObject.search_results = {}
+    ecommerceObject.search_results.event_name = searchResultClicked && searchResult ? 'select_item' : 'view_item_list'
+    ecommerceObject.search_results.term = searchQuery
+    ecommerceObject.search_results.sort = variant
+    ecommerceObject.search_results.results = parseInt(this.getMetaTagContent('govuk:search-result-count'))
+    ecommerceObject.search_results.ecommerce = {}
+    ecommerceObject.search_results.ecommerce.items = []
+
+    if (searchResultClicked && searchResult) {
+      ecommerceObject.search_results.ecommerce.items.push({
+        item_id: searchResult.getAttribute('data-ecommerce-path'),
+        item_name: searchResult.textContent,
+        item_list_name: listTitle,
+        index: this.getIndex(searchResult, Array.prototype.indexOf.call(ecommerceRows, searchResult)) + startPosition
```

Comment:
Can you explain what this line does?

Reply:
As per our discussion earlier, this code has been refactored to be more readable following a suggestion from @AshGDS.